CRM/Utils/Check/Component/Cms.php

```diff
@@ -16,6 +16,43 @@
  */
 class CRM_Utils_Check_Component_Cms extends CRM_Utils_Check_Component {
 
+  /**
+   * For sites running from Composer, ensure public assets have been correctly installed.
+   *
+   * @return CRM_Utils_Check_Message[]
+   */
+  public static function checkPublicAssets(): array {
+    $checks = [];
+    $assetDir = $GLOBALS['civicrm_asset_map']['civicrm/civicrm-core']['dest'] ?? NULL;
+    // No asset directory so this is not a composer-based install
+    if (!$assetDir) {
+      return [];
+    }
+    $errorMsg = NULL;
+    $versionFile = $assetDir . '/js/version.json';
+    if (!file_exists($versionFile)) {
+      $errorMsg = ts('Assets missing from public web directory.');
+    }
+    else {
+      $currentVersion = CRM_Utils_System::version();
+      $assetVersion = json_decode(file_get_contents($versionFile), TRUE);
+      if ($assetVersion !== $currentVersion) {
+        $errorMsg = ts('Wrong version of assets in public web directory: expected %1 but found %2.', [1 => $currentVersion, 2 => $assetVersion]);
+      }
+    }
+    if ($errorMsg) {
+      $checks[] = new CRM_Utils_Check_Message(
+        __FUNCTION__,
+        "$errorMsg Use the command <code>composer civicrm:publish</code> to resolve. " .
```

Comment:
I think a `ts` went missing. (Nobody _should_ be seeing this message to begin with but my magic wand is in the shop.)

Reply:
D'oh now there's no space between the sentence and the Learn More link.

Reply:
ts(' ')

Reply:
ts("😆")